tests/acceptance/helpers/httpHelper.js

```diff
@@ -49,3 +63,261 @@ exports.checkOCSStatus = function (response, message) {
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.move = function (urlData, payload, userId, header) {
+  const options = { ...payload, method: 'MOVE' }
+  return this.requestEndpoint(urlData, options, userId, header)
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.propfind = function (urlData, payload, userId, header) {
+  const options = { ...payload, method: 'PROPFIND' }
+  return this.requestEndpoint(urlData, options, userId, header)
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.report = function (urlData, payload, userId, header) {
+  const options = { ...payload, method: 'REPORT' }
+  return this.requestEndpoint(urlData, options, userId, header)
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.put = function (urlData, payload, userId, header) {
+  const options = { ...payload, method: 'PUT' }
+  return this.requestEndpoint(urlData, options, userId, header)
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.proppatch = function (urlData, payload, userId, header) {
+  const options = { ...payload, method: 'PROPPATCH' }
+  return this.requestEndpoint(urlData, options, userId, header)
+}
+
+/**
+ *
+ * @param {string} urlString
+ *
+ * @returns {string}
+ */
+const getBackEndUrl = function (urlString) {
+  switch (urlString) {
+    case 'globalBackendUrl':
+      return client.globals.backend_url
+    case 'currentBackendUrl':
+      return backendHelper.getCurrentBackendUrl()
+    default:
+      return urlString
+  }
+}
+
+/**
+ *
+ * @param {string} app
+ *
+ * @returns {string}
+ */
+const getAppPath = function (app) {
+  switch (app) {
+    case 'sharing':
+      return 'apps/files_sharing/api/v1/shares'
+    case 'testing':
+      return 'apps/testing/api/v1'
+    case 'cloud':
+      return 'cloud'
+    case 'remote':
+      return 'remote.php'
+  }
+}
+
+/**
+ *
+ * @param {object} urlData
+ * @param {object} payload
+ * @param {string} userId
+ * @param {object} header
+ *
+ * @returns {node-fetch}
+ */
+exports.requestOCSEndpoint = function (urlData, payload, userId = 'admin', header = {}) {
+  const { backendUrlString, app, appParams = '', keyValueArray, queryString = '' } = urlData
+  const backendUrl = getBackEndUrl(backendUrlString || 'globalBackendUrl')
+  const appPath = getAppPath(app)
+  let url
+  if (keyValueArray) {
+    url = new URL(join(backendUrl, '/ocs/v2.php/', appPath))
```

Comment:
IMO 
```node
const something  = '/ocs/v2.php/'
```
would be nicer

Reply:
@kiranparajuli589 `something` doesn't help, It's better if it is the way it is instead of providing a vague variable name. Nice variable name suggestion is welcome, but I think it's fine the way it is.